switcher_one_pot.ino

```diff
@@ -0,0 +1,160 @@
+#include <TM1637Display.h>
+#include <RotaryEncoder.h>
+#include <./LimitedRotaryEncoder.h>
+// Encoder must be included before Control Surface
+#include <Encoder.h>
+#include <Control_Surface.h> // Include the library
+
+#define ROTARYSTEPS 1
+// Last known rotary position.
+//int lastPos = -1;
+int modWheelMsg = 26; //initial CC number
+
+// Variables will change:
+enum State { MODE_LOOP, MODE_SCAN, MODE_CONFIG };
+int current_state = MODE_LOOP;
+
+//enum Potentiometer { bank, bank2 };//nie wiem czy to jest dobrze, chyba wlasnie zrobilem kolizje nazw zmiennych ale zobacz
+int whichPot = 1;
+int oldMsg;
+int oldPot;
+
+const uint8_t SEG_DONE[] = {
+	SEG_A | SEG_D | SEG_E | SEG_F,          // C
+	SEG_B | SEG_C | SEG_E | SEG_F | SEG_G,  // H
+	SEG_B | SEG_C | SEG_D | SEG_E | SEG_F,  // U
+	SEG_B | SEG_C | SEG_D                   // J
+};
+
+TM1637Display display(14, 15);
+
+// Create two MIDI interfaces
+USBMIDI_Interface usbmidi;
+HardwareSerialMIDI_Interface serialmidi = {Serial1, MIDI_BAUD};
+
+// And group them together
+MultiMIDI_Interface<2> midi = {{&usbmidi, &serialmidi}};
+ 
+// Bank with 120 settings in total, increments of 1 address at a time
+Bank<120> bank = {1};
+
+// Setup a RoraryEncoder for pins A2 and A3:
+// dwa obiekty encoder na tych samych pinach i tym samym fixycznym encoderze
+//RotaryEncoder encoder(2, 3);
+LimitedRotaryEncoder encoderSelectMode = {{1, 2}, {2, 3}};  //encoder do wyboru potka
+LimitedRotaryEncoder encoderConfigMode = {{0, 119}, {2, 3}}; //encoder do wyboru midi messaga
```

Comment:
dlaczego dwa encodery - jeden na tryb, po to, zeby kazda instancja pamietala swoja poprzednia pozycje. FGdyby byl tylko jeden obiekt encoder to przy zmianie trybu nadpisze sie ostatnia pozycja. napisalem wiec klase dziedziczaca po `RotaryEncoder` i zawierajaca funkcje `encoderConfig` w metodzie `readInRange` - dzieki temu mam dwa encodery o roznych zakresach (1-n i 0-119) na tym samym fizycznym encoderze. 
problem polega na tym, ze jak widzisz w wykomentowanej funkcji po zczytaniu encodera jest wolany `display` zeby wyswietlic na biezaco nowa wartosc. kiedy display byl poza funkcja dzialalo to za wolno, nie nadazal.
Nooo i klasa `LimitedRotaryEncoder` nie bedzie zawierac tego displaya wiec nie zadziala. 
Teraz - jak to rozwiazac? dorzucic do klasy `LimitedRotaryEncoder` dziedziczenie po `display`? troche to sie wydaje overkill. wyjebac klase i zostawic funkcje ale z wyborem na ktory encoder dziala? tez troche messy, nie wiem jak to ugryzc tak elegancko